### ISSUE TITLE
Améliorer les filtres sur la page des arrêtés (suite)

### ISSUE THREAD

**Issue by johanricher** [MTES-MCT/dialog#1027]
## Contexte

[La page des arrêtés](https://dialog.beta.gouv.fr/regulations) permet de naviguer dans tous les arrêtés dont l'état est "Publié" dans DiaLog (pour tout le monde, y compris hors connexion), et ceux dont l'état est "Brouillon" (seulement pour les utilisateurs membres de l'organisation qui les a créés).

Dans #902, des fonctionnalités de recherche (filtres) ont été ajoutées et améliorées.

En parallèle sur #202, [la page Carte des restrictions](https://dialog.beta.gouv.fr/carte) va proposer des fonctionnalités de recherche plus approfondies sur les restrictions contenues dans les arrêtés, pour les cas d'usage d'utilisation des données. 

La page des arrêtés tend à devenir une partie intégrante de l'espace de travail des agents ["mon espace"](https://dialog.beta.gouv.fr/mon-espace) sur DiaLog, donc avec des fonctionnalités d'administrations orientées pour faciliter ce cas d'usage de création d'arrêtés.

## Exploration

En cours (voir ci-dessous). Le ticket sera mis à jour une fois l'exploration terminée.

**Comment by johanricher** [comment]
Extrait du [CR de la réunion du 22 octobre 2024](https://pad.incubateur.net/#2024-10-22---Point-avec-Mayenne) avec le contact de la commune de Mayenne.

>  Sur la vue tableau des arrêtés il faudrait avoir un filtre par défaut sur notre organisation

Il s'agissait d'un "could have" sur #902 mais écarté car posant des questions UX restées sans réponses et donc une fonctionnalité laissée pour une itération ultérieure.

Si le compte fait partie de plusieurs organisations, laquelle est sélectionnée par défaut ? Il faudrait potentiellement la possibilité de faire un multi-select dans ce menu (e.g . si je fais partie de la MEL, je veux voir dans la même liste les arrêtés de la MEL mais aussi des communes la composant et que ce soit le filtre par défaut).

> pouvoir aussi faire une recherche d'un nom d'organisation dans la dropdown "Organisation" pour en chercher une dans la liste

Là aussi un "could have" de #902 pour des raisons plutôt techniques cette fois-ci.

Ici il s'agirait d'implémenter un menu select (voire multi-select) avec recherche plein texte et autocomplétion. Voir exemple d'UI / UX sur [data.gouv.fr](https://www.data.gouv.fr/fr/datasets) (sans multi-select néanmoins) :

[Screencast from 2024-10-23 14-20-53.webm](https://github.com/user-attachments/assets/5157e1da-27f3-449c-8fc3-6312972294e4)

Pourrait-on avoir un éclairage sur la faisabilité technique sur la stack DiaLog ?

Autre amélioration que j'avais proposé sur #902 que je remets dans ce lot à explorer pour itérer :

Retirer du menu "Organisation" toutes les organisations qui n'ont pas au moins un arrêté dont l'état est "Publié" dans DiaLog (cela encombre ce menu pour rien car filtrer sur une organisation sans arrêté ne donnera aucun résultat).

**Comment by florimondmanca** [comment]
> Pourrait-on avoir un éclairage sur la faisabilité technique sur la stack DiaLog ?

Pour un filtre type "Menu avec recherche plein texte et autocomplétion", on a quelque chose qui se rapproche au niveau des champs Commune et Voie dans le formulaire de voie nommée.

Il y a une différence que je vois, qui serait d'afficher la liste des options quand on clique sur le champ (comme un select), alors que pour Commune / Voie on n'affiche rien tant qu'on n'a pas fait de recherche

Mais c'est du paramétrage, on pourrait quand même réutiliser l'essentiel de notre composant autocomplete, je pense

> Retirer du menu "Organisation" toutes les organisations qui n'ont pas au moins un arrêté dont l'état est "Publié" dans DiaLog (cela encombre ce menu pour rien car filtrer sur une organisation sans arrêté ne donnera aucun résultat).

Bien vu

**Comment by florimondmanca** [comment]
> Retirer du menu "Organisation" toutes les organisations qui n'ont pas au moins un arrêté dont l'état est "Publié" dans DiaLog (cela encombre ce menu pour rien car filtrer sur une organisation sans arrêté ne donnera aucun résultat).

La seule subtilité c'est qu'on peut aussi voir les brouillons quand on est connecté, et si on voulait être le + fidèle on ne montrerait que les organisations dont on peut voir les brouillons (= celles dont on fait partie)... Mais ça semble devenir un peu compliqué

Est-ce que ce "besoin" ne serait pas levé par le menu + autocomplétion ?

**Comment by aureliebaton** [comment]
J'avais pensé à une autre option plus simple que le multi-select pour les organisations (je pense) :
Avoir dans la liste déroulante une entrée "Mes organisations" sélectionnée par défaut.
L'idée principale étant que quand je me connecte, je ne veux avoir une vue des arrêtés de mon ou mes organisations (si j'en ai plusieurs). 

Dans la liste j'aurais donc par exemple :
* Mes organisations 
* Toutes les organisations
* Organisation A
* Organisation B
* Organisation C
....
---
La vraie question plus long terme étant : une fois qu'on aura la vue linsting des restrictions, est-ce qu'on restreindra la vue "arrêtés" à simplement mes organisations (parce que à la base on avait fait ça juste en attendant d'avoir une vraie vue listing". Et je pense que ça ne fera pas vraiment de sens de garder la vue de tous les arrêtés de tout le monde.

**Comment by aureliebaton** [comment]
> Est-ce que l'entrée `Toutes les organisations` reste utile dans le cas où on peut sélectionner ses orgas ou une de manière spécifique ?

Je suis d'accord qu'en mode connecté y en a pas besoin, on pourrait la retirer.

La question est, est-ce qu'on garde "toutes les organisation" en mode déconnecté en attendant d'avoir la vue "listing" ? @MathieuFV @johanricher

**Comment by florimondmanca** [comment]
Je dirais que ne pas avoir cette valeur "Toutes les organisations" pourrait être confusant, on risquerait de ne trouver aucun résultat uniquement parce que le filtre a été mis sur une organisation particulière

D'ailleurs en mode déconnecté s'il cette option n'est pas là, qu'est-ce qui est sélectionné dans le filtre "organisation" ?

**Comment by aureliebaton** [comment]
> Je dirais que ne pas avoir cette valeur "Toutes les organisations" pourrait être confusant, on risquerait de ne trouver aucun résultat uniquement parce que le filtre a été mis sur une organisation particulière
> 
> D'ailleurs en mode déconnecté s'il cette option n'est pas là, qu'est-ce qui est sélectionné dans le filtre "organisation" ?

Disons qu'on peut le garder maintenant, et qu'on verra plus tard quand on aura la vue listing (à ce moment là on en aura à priori plus besoin)

**Comment by florimondmanca** [comment]
Je clos car ça a été déplacé vers "Terminé" mais sans fermer le ticket
